functional test / smooth switch : use currentTime instead of readyState as test condition

mangui opened the pull request:
### Description of the Changes

try to workaround readyState not being reliable on Saucelabs VM

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] no commits have been done in dist folder (we will take care of updating it)
- [x] new unit / functional tests have been added (whenever applicable)
- [?] Travis tests are passing (or test results are not worse than on master branch :))
- [x] API or design changes are documented in API.md

tchakabam commented on tests/functional/auto/setup.js:
```diff
@@ -159,13 +159,19 @@ describe('testing hls.js playback in the browser on "' + browserDescription + '"
-        }, 12000);
+        window.hls.on(window.Hls.Events.LEVEL_SWITCHED, function (event, data) {
+          var currentTime = video.currentTime;
+          if (data.level === window.hls.levels.length - 1) {
```
should we return a `false` code when the level is not the one we expect? i.e handle the `else` condition here too?

mangui replied:
actually we receive that event twice : once first fragment played, and another time when effectively smooth switching

mangui commented:
[interesting failure on IE11](https://travis-ci.org/video-dev/hls.js/jobs/353191445)

```
2018-03-14 07:03:51.303: "[log] > switching to level 4"
2018-03-14 07:03:51.304: "[log] > main stream:IDLE->WAITING_LEVEL"
2018-03-14 07:03:51.304: "[log] > target start position:0"
2018-03-14 07:03:51.305: "[log] > target start position not buffered, seek to buffered.start(0) 0.0100907"
2018-03-14 07:03:51.305: "[log] > adjust currentTime from 0 to 0.0100907"
2018-03-14 07:03:51.306: "[log] > Updating Media Source duration to 634.584"
2018-03-14 07:03:51.306: "[log] > media seeking to 0.000"
2018-03-14 07:03:51.307: "[log] > media seeked to 0.000"
2018-03-14 07:03:51.383: "[log] > level 4 loaded [0,63],duration:634.567"
2018-03-14 07:03:51.384: "[log] > main stream:WAITING_LEVEL->IDLE"
2018-03-14 07:03:51.384: "[log] > Loading 1 of [0 ,63],level 4, currentTime:0.000,bufferEnd:10.000"
2018-03-14 07:03:51.386: "[log] > main stream:IDLE->FRAG_LOADING"
2018-03-14 07:03:51.654: "[log] > set nextLevel:4"
2018-03-14 07:03:51.975: "[log] > Loaded  1 of [0 ,63],level 4"
2018-03-14 07:03:51.975: "[log] > main stream:FRAG_LOADING->PARSING"
2018-03-14 07:03:51.976: "[log] > Parsing 1 of [0 ,63],level 4, cc 0"
2018-03-14 07:03:51.976: "[log] > main:switch detected"
2018-03-14 07:03:52.335: "[log] > main track:audio,container:audio/mp4,codecs[level/parsed]=[mp4a.40.5/mp4a.40.5]"
2018-03-14 07:03:52.336: "[log] > main track:video,container:video/mp4,codecs[level/parsed]=[avc1.640028/avc1.640028]"
2018-03-14 07:03:52.365: "[log] > Parsed audio,PTS:[10.018,20.002],DTS:[10.018/20.002],nb:430,dropped:0"
2018-03-14 07:03:52.558: "[log] > Parsed video,PTS:[10.033,20.017],DTS:[10.000/20.000],nb:600,dropped:0"
2018-03-14 07:03:52.603: "[log] > main stream:PARSING->PARSED"
2018-03-14 07:03:52.628: "[log] > main buffered : [0.010,10.018]"
2018-03-14 07:03:52.628: "[log] > latency/loading/parsing/append/kbps:267/304/647/25/84842"
2018-03-14 07:03:52.628: "[log] > main stream:PARSED->IDLE"
2018-03-14 07:03:52.629: "[log] > SN just loaded, load next one: 2"
2018-03-14 07:03:52.629: "[log] > Loading 2 of [0 ,63],level 4, currentTime:0.768,bufferEnd:10.018"
2018-03-14 07:03:52.631: "[log] > main stream:IDLE->FRAG_LOADING"
2018-03-14 07:03:52.960: "[log] > Loaded  2 of [0 ,63],level 4"
2018-03-14 07:03:52.960: "[log] > main stream:FRAG_LOADING->PARSING"
2018-03-14 07:03:52.960: "[log] > Parsing 2 of [0 ,63],level 4, cc 0"
2018-03-14 07:03:53.244: "[log] > Parsed audio,PTS:[20.002,30.010],DTS:[20.002/30.010],nb:431,dropped:0"
2018-03-14 07:03:53.245: "[log] > Parsed video,PTS:[20.033,30.017],DTS:[20.000/30.000],nb:600,dropped:0"
2018-03-14 07:03:53.245: "[log] > main stream:PARSING->PARSED"
2018-03-14 07:03:53.300: "[log] > main buffered : [0.010,10.018]"
2018-03-14 07:03:53.302: "[log] > latency/loading/parsing/append/kbps:153/169/294/55/77579"
2018-03-14 07:03:53.302: "[log] > main stream:PARSED->IDLE"
2018-03-14 07:03:53.302: "[log] > Loading 1 of [0 ,63],level 4, currentTime:0.982,bufferEnd:10.018"
2018-03-14 07:03:53.339: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.018,bufferEnd:10.018"
2018-03-14 07:03:53.420: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.521: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.606: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.675: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.770: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.855: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.077,bufferEnd:10.018"
2018-03-14 07:03:53.961: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.139,bufferEnd:10.018"
2018-03-14 07:03:54.050: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.228,bufferEnd:10.018"
2018-03-14 07:03:54.155: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.333,bufferEnd:10.018"
2018-03-14 07:03:54.279: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.458,bufferEnd:10.018"
2018-03-14 07:03:54.365: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.544,bufferEnd:10.018"
2018-03-14 07:03:54.455: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.633,bufferEnd:10.018"
2018-03-14 07:03:54.546: "[log] > Loading 1 of [0 ,63],level 4, currentTime:1.725,bufferEnd:10.018"
```

might be worth checking before tagging a release
